uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -379,9 +455,14 @@ define([
      */
     function initListeners () {
       listeners.push(
-        $rootScope.$on('LeaveRequestPopup::updateBalance', function () {
-          vm.updateBalance();
-          setDaySelectionMode();
+        $rootScope.$on('LeaveRequestPopup::updateBalance', function (event, absenceTypes) {
```

Comment:
the fact that you are receiving a bunch of absence types from an event about the "balance" is really hard to make sense of, this can't be kept like this, one thing really doesn't follow the other

Reply:
Well, absence types contain balance... I will try to come up with something better.

Reply:
I changed the name of the variable to "absenceTypesWithBalances"

Reply:
slightly related: you chaged `::updateBalance` to `::updateEntitlements`, but it was correct how it was before. You are showing the current balance of the absence types, not the original entitlements

Reply:
Fixed.